Source/Particles/PhysicalParticleContainer.cpp

```diff
@@ -341,51 +347,56 @@ PhysicalParticleContainer::AddPlasmaFromFile(const std::string s_f,
+
 #   if (defined WARPX_DIM_3D) || (defined WARPX_DIM_XZ)
-        auto const npart = ps.second["position"]["x"].getExtent()[0];
-        std::shared_ptr<amrex::ParticleReal> ptr_x = ps.second["position"]["x"].loadChunk<amrex::ParticleReal>();
-        double const position_unit_x = ps.second["position"]["x"].unitSI();
-        std::shared_ptr<amrex::ParticleReal> ptr_z = ps.second["position"]["z"].loadChunk<amrex::ParticleReal>();
-        double const position_unit_z = ps.second["position"]["z"].unitSI();
-        std::shared_ptr<amrex::ParticleReal> ptr_ux = ps.second["momentum"]["x"].loadChunk<amrex::ParticleReal>();
-        double const momentum_unit_x = ps.second["momentum"]["x"].unitSI();
-        std::shared_ptr<amrex::ParticleReal> ptr_uz = ps.second["momentum"]["z"].loadChunk<amrex::ParticleReal>();
-        double const momentum_unit_z = ps.second["momentum"]["z"].unitSI();
+        auto const npart = ps["position"]["x"].getExtent()[0];
+        std::shared_ptr<amrex::ParticleReal> ptr_x = ps["position"]["x"].loadChunk<amrex::ParticleReal>();
+        double const position_unit_x = ps["position"]["x"].unitSI();
+        std::shared_ptr<amrex::ParticleReal> ptr_z = ps["position"]["z"].loadChunk<amrex::ParticleReal>();
+        double const position_unit_z = ps["position"]["z"].unitSI();
+        std::shared_ptr<amrex::ParticleReal> ptr_ux = ps["momentum"]["x"].loadChunk<amrex::ParticleReal>();
+        double const momentum_unit_x = ps["momentum"]["x"].unitSI();
+        std::shared_ptr<amrex::ParticleReal> ptr_uz = ps["momentum"]["z"].loadChunk<amrex::ParticleReal>();
+        double const momentum_unit_z = ps["momentum"]["z"].unitSI();
 #   else
         amrex::Abort("AddPlasmaFromFile is only implemented for 2D and 3D\n");
 #   endif
 #   if (defined WARPX_DIM_3D)
-        std::shared_ptr<amrex::ParticleReal> ptr_y = ps.second["position"]["y"].loadChunk<amrex::ParticleReal>();
-        double const position_unit_y = ps.second["position"]["y"].unitSI();
-        std::shared_ptr<amrex::ParticleReal> ptr_uy = ps.second["momentum"]["y"].loadChunk<amrex::ParticleReal>();
-        double const momentum_unit_y = ps.second["momentum"]["y"].unitSI();
+        std::shared_ptr<amrex::ParticleReal> ptr_y = ps["position"]["y"].loadChunk<amrex::ParticleReal>();
+        double const position_unit_y = ps["position"]["y"].unitSI();
+        std::shared_ptr<amrex::ParticleReal> ptr_uy = ps["momentum"]["y"].loadChunk<amrex::ParticleReal>();
+        double const momentum_unit_y = ps["momentum"]["y"].unitSI();
 #   endif
-        series.flush();
-
-        mass=p_m*mass_unit;
-        charge=p_q*charge_unit;
+        series->flush();
 
         amrex::Real weight;
-        if (q_tot != 0.0){
-            weight = q_tot/(p_q*amrex::Real(npart));
+        if (q_tot != 0.0) {
+            weight = std::abs(q_tot) / ( std::abs(charge) * amrex::Real(npart) );
+            if (ps.contains("weighting"))
```

Comment:
Perfect! Thank you for adding this @ax3l !! :)
```suggestion
            if (ps.contains("weighting")) {
```